test/e2e_node/node_shutdown_linux_test.go

```diff
@@ -86,18 +88,18 @@ var _ = SIGDescribe("GracefulNodeShutdown [Serial] [NodeAlphaFeature:GracefulNod
 			framework.ExpectNoError(err)
 			framework.ExpectEqual(len(list.Items), len(pods), "the number of pods is not as expected")
 
+			ginkgo.By("Verifying batch pods are running")
 			for _, pod := range list.Items {
```

Comment:
this is strange. `CreateBatch` is already checking for readiness, is it?

Reply:
It was a bit tricky, here's what I found:

* [`CreateBatch`](https://github.com/kubernetes/kubernetes/blob/bd2e557/test/e2e/framework/pods.go#L111) calls [`CreateSync`](https://github.com/kubernetes/kubernetes/blob/bd2e557/test/e2e/framework/pods.go#L119) for each pod
* CreateSync polls if the pod is ready and checks that error is nil from [`e2epod.WaitTimeoutForPodReadyInNamespace`](https://github.com/kubernetes/kubernetes/blob/bd2e557/test/e2e/framework/pods.go#L103)
* `WaitTimeoutForPodReadyInNamespace` polls `podRunningAndReady`
* [`podRunningAndReady`](https://github.com/kubernetes/kubernetes/blob/bd2e557/test/e2e/framework/pod/resource.go#L191-L193) checks if the pod status phase is running AND that the pod is ready

In our case, the pods were in crash loop backoff and failing to start due to the sleep infinity issue.

The problem is that pod is not ready `podRunningAndReady` returns (false, nil) (where nil is the error). Higher up the stack `WaitTimeoutForPodReadyInNamespace` will poll until the timeout, but only fail the test if the error is not nil. 
 
In the current case, the pods were in crash loop backoff and were not ready. So, basically `podRunningAndReady` kept checking that the pods got into ready state (which they were not) so eventually the `WaitTimeoutForPodReadyInNamespace` hit the timeout, and ended up continuing (since the error was nil). 

With the added check here we will ensure that if the pods fail getting into ready=true condition the test will fail early rather the continuing (and thus failing due to a different issue)